vochain/state/state.go

```diff
@@ -420,6 +430,32 @@ func (v *State) SetHeight(height uint32) {
 	v.currentHeight.Store(height)
 }
 
+// SetTimestamp sets the timestamp for the current (not committed) block.
+func (v *State) SetTimestamp(timestamp uint32) error {
```

Comment:
We should probably check for uint32 overflows here too

Reply:
By overflow you mean calling SetTimestamp with a uint32 value higher than  2^32 -1 ? If that is what you mean, there is no way to check the overflow here, the uint32 will already have an "overflow" value at this point. I.e imagine I call `SetTimestamp(-1)` (underflow), the function will receive timestamp=2^32 -1 which is a valid uint32

Reply:
You're right. I was thinking about handling possible unwanted values, more of a design consideration.
The responsibility basically lies in ensuring that the function is used correctly.
My thoughts were on modifying the function to something like:

```golang
func TimestampToTime(timestamp uint32) (time.Time, error) {
    if timestamp < 0 || timestamp > math.MaxUint32 {
        // return error
    }
    return time.Unix(timestamp, 0), nil
}
```

But it does not make much sense, Thank  you for the clarifications.